File: low-level-design/design-patterns/python/behavioural/chain-of-responsibility-pattern/security_check_handler.py
from handler import BaseHandler,Handler,DeploymentRequest

class SecurityCheckHandler(BaseHandler):
    def __init__(self,handler:Handler):
        self.set_next(handler)
    
    def handle(self,request:DeploymentRequest):
        if not request.package_secure:
            print("SecurityCheckHandler: Package not secure. Denying deployment.")
            return False
        print("SecurityCheckHandler: Package security check passed. Proceeding to next handler.")
        return self.call_next(request)

File: low-level-design/design-patterns/golang/behavioural/chain-of-responsibility-pattern/cor/security_check_handler.go
package cor

import (
	"errors"
	"fmt"
)

func NewSecurityCheckHandler(next Handler) *SecurityCheckHandler {
	handler := &SecurityCheckHandler{}
	handler.SetNext(next)
	return handler
}

type SecurityCheckHandler struct {
	BaseHandler
}

func (h *SecurityCheckHandler) Handle(request *DeploymentRequest) error {
	if !request.PackageSecure {
		fmt.Printf("Security check failed for service %s\n", request.ServiceName)
		return errors.New("security check failed")
	}
	fmt.Printf("Security check passed for service %s\n", request.ServiceName)
	return h.CallNext(request)
}
